math/minuit2/src/MnUserParameterState.cxx

```diff
@@ -287,6 +288,43 @@ void MnUserParameterState::Add(const std::string &name, double val)
       SetValue(name, val);
 }
 
+void MnUserParameterState::AddCovariance(const MnUserCovariance &cov)
+{
+
+   unsigned int nrow = VariableParameters();
+   assert(cov.Nrow() >= nrow);
+
+   // add external covariance matrix
+   fCovariance = cov;
+
+   // compute and add internal covariance matrix
+   MnUserCovariance covsqueezed;
+   if (cov.Nrow() > nrow)
+      covsqueezed = MnCovarianceSqueeze()(cov, nrow);
+   else if (cov.Nrow() == nrow)
+      covsqueezed = cov;
+
+   MnAlgebraicVector params(nrow);
+   for (unsigned int i = 0; i < nrow; i++)
+      params(i) = fParameters.Params()[i];
+
+   MnAlgebraicSymMatrix covmat(nrow);
+   for (unsigned int i = 0; i < nrow; i++)
+      for (unsigned int j = i; j < nrow; j++)
+         covmat(i, j) = covsqueezed(i, j);
+
+   fIntCovariance = fParameters.Trafo().Ext2intCovariance(params, covmat);
+   fIntCovariance.Scale(0.5); //?
+   fCovarianceValid = true;
+   fCovStatus = 0; //?
+}
+
+void MnUserParameterState::AddCovariance(std::span<const double> covvec, unsigned int nrow)
+{
+   MnUserCovariance cov(covvec, nrow);
+   MnUserParameterState::AddCovariance(cov);
+}
```

Comment:
```suggestion
```
Similar to the other comment: what is the added benefit of this overload? Usually, users don't use Minuit 2 directly anymore, and IMHO it's also no problem to do `MnUserParameterState::AddCovariance({covvec, nrow})`, implicitly constructing the `MnConvarianceMatrix`. Only two characters more ;)